compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -1360,18 +1360,24 @@ fn early_bound_lifetimes_from_generics<'a, 'tcx: 'a>(
 fn predicates_defined_on(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredicates<'_> {
     let mut result = tcx.explicit_predicates_of(def_id);
     debug!("predicates_defined_on: explicit_predicates_of({:?}) = {:?}", def_id, result,);
-    let inferred_outlives = tcx.inferred_outlives_of(def_id);
+    let inferred_outlives: Vec<_> = tcx
+        .inferred_outlives_of(def_id)
+        .iter()
+        .map(|(clause, span)| {
+            (ty::Binder::dummy(ty::PredicateKind::Clause(*clause)).to_predicate(tcx), *span)
+        })
+        .collect();
```

Comment:
I don't like that we need to do this

Reply:
Can you use `ExactSizeIterator` maybe? Or `Peekable::peek`?

Reply:
Fixed by creating the iterator inside the check.